platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/SslLdapLoginModule.java

```diff
@@ -386,60 +402,53 @@ protected BundleContext getContext() {
-                        ref);
+        try {
+            ldapConnectionFactory = createLdapConnectionFactory(connectionURL, startTls);
+        } catch (LdapException e) {
+            LOGGER.info(
+                    "Unable to create LDAP Connection Factory. LDAP log in will not be possible.",
+                    e);
+        }
+    }
 
-                if (encryptionService != null) {
-                    String decryptedPassword = encryptionService.decryptValue(
-                            option.get(CONNECTION_PASSWORD));
-                    option.put(CONNECTION_PASSWORD, decryptedPassword);
-                } else {
-                    LOGGER.info("Encryption service reference for ldap was null.");
-                }
+    @Override
+    public boolean login() throws LoginException {
+        boolean isLoggedIn = false;
+        String message = "";
+        try {
+            isLoggedIn = doLogin();
 
-            } catch (SecurityException | IllegalStateException e) {
-                LOGGER.info(
-                        "Error decrypting connection password passed into ldap configuration: ", e);
-            } finally {
-                if (ref != null) {
-                    bundleContext.ungetService(ref);
-                }
-            }
-            connectionURL = (String) options.get(CONNECTION_URL);
-            connectionUsername = (String) options.get(CONNECTION_USERNAME);
-            connectionPassword = (String) options.get(CONNECTION_PASSWORD);
-            userBaseDN = (String) options.get(USER_BASE_DN);
-            userFilter = (String) options.get(USER_FILTER);
-            userSearchSubtree = Boolean.parseBoolean((String) options.get(USER_SEARCH_SUBTREE));
-            roleBaseDN = (String) options.get(ROLE_BASE_DN);
-            roleFilter = (String) options.get(ROLE_FILTER);
-            roleNameAttribute = (String) options.get(ROLE_NAME_ATTRIBUTE);
-            roleSearchSubtree = Boolean.parseBoolean((String) options.get(ROLE_SEARCH_SUBTREE));
-            startTls = Boolean.parseBoolean(String.valueOf(options.get(SSL_STARTTLS)));
-            bindMethod = (String) options.get(BIND_METHOD);
-            realm = (String) options.get(REALM);
-            kdcAddress = (String) options.get(KDC_ADDRESS);
-
-            if (ldapConnectionFactory != null) {
-                ldapConnectionFactory.close();
-            }
+            message = "Username [" + user
+                    + "] could not log in successfuly using LDAP authentication due to an exception";
 
-            try {
-                ldapConnectionFactory = createLdapConnectionFactory(connectionURL, startTls);
-            } catch (LdapException e) {
-                LOGGER.info(
-                        "Unable to create LDAP Connection Factory. LDAP log in will not be possible.",
-                        e);
+            if (!isLoggedIn) {
+                SecurityLogger.audit("Username [" + user + "] failed LDAP authentication.");
```

Comment:
`SecurityLogger` now has a method `cleanAndEncode` that scrubs newlines from the message to prevent log injection attacks. So we should be good on that score.